pick.lic

```diff
@@ -2,523 +2,665 @@
+      dismantle(box)
+    when 'Your hands are too full for that'
+      stow_hands_except(box['noun'])
+      dismantle(box)
+    when 'You can not dismantle that'
+      DRCI.dispose_trash(box['noun'], @worn_trashcan, @worn_trashcan_verb)
     end
   end
 
-  def check_danger
-    waitrt?
-    pause 0.5 while stunned?
-    echo " bleeding?: #{bleeding?}" if UserVars.lockpick_debug
-    echo " checkpoison: #{checkpoison}" if UserVars.lockpick_debug
-    return unless bleeding? || checkpoison
+  def loot(box_noun)
+    echo "Looting #{box_noun}..." if @debug
+    if DRC.bput("open my #{box_noun}", /You open/, /^In the .* you see .*\./, 'That is already open', 'It is locked') == 'It is locked'
+      echo 'Bug: Tried to loot locked box...'
+      return
+    end
 
-    snapshot = Room.current.id
-    wait_for_script_to_complete('safe-room')
-    walk_to(snapshot)
+    fill_result = DRC.bput("fill my #{@settings.gem_pouch_adjective} pouch with my #{box_noun}", 'You fill your', 'You open your', 'What were you referring to', /any gems/, /too full to fit/)  if (@settings.fill_pouch_with_box || @settings.loot_specials.empty?)
+    case fill_result
+    when /too full to fit/
+      swap_out_full_gempouch
+    end
+
+    box_items = DRCI.get_item_list(box_noun, 'look');
+    loot = box_items.map { |item| DRC.get_noun(item) }
+    echo "box items: #{box_items} - loot: #{loot}" if @debug
+
+    loot.each { |item| loot_item(item, box_noun) }
   end
 
-  def disarm_speed?(box, speed)
-    waitrt?
-    echo "disarm_speed?(#{box}, #{speed})" if UserVars.lockpick_debug
-    Flags.reset('disarm-more')
+  def loot_item(item, box)
+    # We don't touch glowing fragments, since we took off all our armor...
+    return if item =~ /fragment/i
 
-    case bput("disarm my #{box} #{speed}", @disarm_failed, @disarm_retry, 'Roundtime')
-    when *@disarm_failed
-      beep
-      beep
-      echo('**SPRUNG TRAP**')
+    item_long = nil
+    case DRC.bput("get #{item} from my #{box}", 'You get (.*) from inside', 'You pick up \d* \w* (?:lirum|dokora|kronar)s?')
+    when /You pick up/ # Coins
+      return
+    when /You get (.*) from inside/
+      item_long = Regexp.last_match(1)
+      echo "Looted: #{item_long}" if @debug
+    end
 
-      check_danger
+    if @settings.loot_specials.find { |x| /\b#{x['name']}\b/i =~ item_long }
+      DRCI.put_away_item?(item, special['bag'])
+      return
+    end
 
-      return false
-    when *@disarm_retry
-      new_speed = reget(10, 'something to shift') ? 'careful' : speed
-      return disarm_speed?(box, new_speed)
+    if @loot_nouns.find { |thing| item_long.include?(thing) && !item_long.include?('sunstone runestone') }
+      case DRC.bput("stow my #{item}", 'You put', 'You open', 'is too full to fit another gem', /You'd better tie it up before putting/)
+      when /You put/, /You open/
+        return
+      when /You'd better tie it up/, 'is too full to fit another gem'
```

Comment:
💅 For consistency, since some of the match strings are explcitly expressed as regex (and any string passed to bput will be converted to a regex anyways), suggesting that this section use the explicit regex syntax for all the patterns.
```suggestion
      case DRC.bput("stow my #{item}", /You put/, /You open/, /is too full to fit another gem/, /You'd better tie it up before putting/)
      when /You put/, /You open/
        return
      when /You'd better tie it up/, /is too full to fit another gem/
```